src/duke/main/Duke.java

```diff
@@ -0,0 +1,223 @@
+            System.out.println(Integer.toString(j + 1) + list.get(j));
+        }
+        System.out.println(LINEBREAK);
+    }
+    public static void deleteTask(String line){
+        int index;
+        try{
+            index = Integer.parseInt(line.split(" ")[1]) - 1;
+            if (index >= taskCount || index < 0){
+                throw new DukeException("Error: Task Index is out of bounds.");
+            }
+        }catch (NumberFormatException e){
+            System.out.println(LINEBREAK);
+            System.out.println(" Error: Invalid index (Not an integer).");
+            System.out.println(LINEBREAK);
+            return;
+        }catch (DukeException e){
+            System.out.println(LINEBREAK);
+            System.out.println(e.getMessage());
+            System.out.println(LINEBREAK);
+            return;
+        }
+        System.out.println(LINEBREAK);
+        System.out.println(" I have removed this task:");
+        System.out.println(list.get(index));
+        taskCount--;
+        System.out.println(" Total number of tasks now: " + taskCount);
+        System.out.println(LINEBREAK);
+        list.remove((index));
+        saveTasks();
+    }
+    public static void updateMarkTask(String line, boolean mark){
+        int index;
+        try{
+            index = Integer.parseInt(line.split(" ")[1]) - 1;
+            if (index >= taskCount || index < 0){
+                throw new DukeException("Error: Task Index is out of bounds.");
+            }
+        }catch (NumberFormatException e){
+            System.out.println(LINEBREAK);
+            System.out.println(" Error: Invalid index (Not an integer).");
+            System.out.println(LINEBREAK);
+            return;
+        }catch (DukeException e){
+            System.out.println(LINEBREAK);
+            System.out.println(e.getMessage());
+            System.out.println(LINEBREAK);
+            return;
+        }
+        System.out.println(LINEBREAK);
+        if (mark){
+            System.out.println(" Nice! I've marked this task as done:");
+            System.out.println("  [X] " + list.get(index).getName());
+        }else{
+            System.out.println(" OK, I've marked this task as not done yet:");
+            System.out.println("  [ ] " + list.get(index).getName());
+        }
+        System.out.println(LINEBREAK);
+        list.get(index).setMarked(mark);
+        saveTasks();
+    }
+    public static void addNewTask(String line){
+        String taskType = line.split(" ")[0];
+        if (line.length() > taskType.length()){
+            line = line.substring(taskType.length() + 1);
+        }else{
+            line = "";
+        }
+        try {
+            switch (taskType) {
+            case "todo":
+                if (line.equals("")){
+                    throw new DukeException("Error: Argument of todo should not be empty.");
+                }
+                list.add(new Todo(line, false));
+                break;
```

Comment:
You might want to extract this chunk into a new method for SLAP.

Same issue for the other `case` blocks as well.